src/Validation/CaValidation.php

```diff
@@ -59,3 +108,101 @@ public static function personId($check)
         throw new NotImplementedException(__d('localized', '%s Not implemented yet.'));
     }
 }
+
+class InvalidNumberRule
```

Comment:
Why did you implement this now? The previous PR was good to go, but now you introduce a lot of debatable code pieces here.